subsys/bluetooth/services/ots/ots_olcp.c

```diff
@@ -156,54 +156,37 @@ static enum bt_gatt_ots_olcp_res_code olcp_proc_execute(
 	}
 };
 
-static enum bt_gatt_ots_olcp_res_code olcp_command_decode(
-	const uint8_t *buf, struct bt_gatt_ots_olcp_proc *proc)
+static int olcp_command_decode(const uint8_t *buf, uint16_t len,
+			       struct bt_gatt_ots_olcp_proc *proc)
 {
 	memset(proc, 0, sizeof(*proc));
 
 	proc->type = *buf++;
+	len -= OLCP_PROC_TYPE_SIZE;
 
 	switch (proc->type) {
 	case BT_GATT_OTS_OLCP_PROC_FIRST:
 	case BT_GATT_OTS_OLCP_PROC_LAST:
 	case BT_GATT_OTS_OLCP_PROC_PREV:
 	case BT_GATT_OTS_OLCP_PROC_NEXT:
-		break;
+		if (len != 0) {
+			return -EBADMSG;
+		}
+
+		return 0;
 	case BT_GATT_OTS_OLCP_PROC_GOTO:
+		if (len != BT_GATT_OTS_OLCP_GOTO_PARAMS_SIZE) {
+			return -EBADMSG;
+		}
 		proc->goto_params.id = sys_get_le48(buf);
-		break;
-	default:
-		LOG_WRN("OLCP unsupported procedure type");
-		return BT_GATT_OTS_OLCP_RES_PROC_NOT_SUP;
-	}
-
-	return BT_GATT_OTS_OLCP_RES_SUCCESS;
-}
 
-static bool olcp_command_len_verify(enum bt_gatt_ots_olcp_proc_type type,
-				    uint16_t len)
-{
-	uint16_t ref_len = OLCP_PROC_TYPE_SIZE;
-
-	switch (type) {
-	case BT_GATT_OTS_OLCP_PROC_FIRST:
-	case BT_GATT_OTS_OLCP_PROC_LAST:
-	case BT_GATT_OTS_OLCP_PROC_PREV:
-	case BT_GATT_OTS_OLCP_PROC_NEXT:
-	case BT_GATT_OTS_OLCP_PROC_REQ_NUM_OBJS:
-	case BT_GATT_OTS_OLCP_PROC_CLEAR_MARKING:
-		break;
-	case BT_GATT_OTS_OLCP_PROC_GOTO:
-		ref_len += BT_OTS_OBJ_ID_SIZE;
-		break;
-	case BT_GATT_OTS_OLCP_PROC_ORDER:
-		ref_len += sizeof(uint8_t);
-		break;
+		return 0;
 	default:
-		return true;
+		break;
 	}
 
-	return (len == ref_len);
+	LOG_WRN("OLCP unsupported procedure type");
```

Comment:
We should log the procedure type here

Reply:
I have added the same logging to the OACP file.